core/services/functions/connector_handler.go

```diff
@@ -160,6 +200,105 @@ func (h *functionsConnectorHandler) handleSecretsSet(ctx context.Context, gatewa
 	h.sendResponseAndLog(ctx, gatewayId, body, response)
 }
 
+func (h *functionsConnectorHandler) handleHeartbeat(ctx context.Context, gatewayId string, requestBody *api.MessageBody, fromAddr ethCommon.Address) {
+	var request *OffchainRequest
+	err := json.Unmarshal(requestBody.Payload, &request)
+	if err != nil {
+		h.sendResponseAndLog(ctx, gatewayId, requestBody, HeartbeatResponse{
+			Status:        RequestStateInternalError,
+			InternalError: fmt.Sprintf("failed to unmarshal request: %v", err),
+		})
+		return
+	}
+	if !bytes.Equal(request.RequestInitiator, fromAddr.Bytes()) {
+		h.sendResponseAndLog(ctx, gatewayId, requestBody, HeartbeatResponse{
+			Status:        RequestStateInternalError,
+			InternalError: "requestInitiator doesn't match sender",
+		})
+		return
+	}
+
+	internalId := InternalId(fromAddr.Bytes(), request.RequestId)
+	request.RequestId = internalId[:]
+	h.lggr.Infow("handling offchain heartbeat", "messageId", requestBody.MessageId, "id", internalId, "sender", requestBody.Sender)
```

Comment:
I don't think the other proposed change is correct. In Heartbeat's case, the Heartbeat Daemon will log all internal IDs so it will be easy to match to node logs.